menuinst/utils.py

```diff
@@ -136,49 +136,121 @@ def add_xml_child(parent: XMLTree.Element, tag: str, text: Optional[str] = None)
 
 
 class WinLex:
+    # Note that there are more characters, for example parenthesis,
+    # but in our case parenthesis are required to be quoted for example with
+    # python -c "print('foo')"
+    _META_CHARS = (">", "<", "|", "&")
+
     @classmethod
     def quote_args(cls, args: Sequence[str]):
         # cmd.exe /K or /C expects a single string argument and requires
         # doubled-up quotes when any sub-arguments have spaces:
         # https://stackoverflow.com/a/6378038/3257826
         if (
             len(args) > 2
-            and ("CMD.EXE" in args[0].upper() or "%COMSPEC%" in args[0].upper())
+            and cls._is_cmd(args[0])
             and (args[1].upper() == "/K" or args[1].upper() == "/C")
```

Comment:
You are using `quote_args` now in a function that prepends the script with `cmd.exe /D /K`, so `args[1]` will be `/D` even though `/K` is added. Will this cause any issues? Does the check need to be more robust to parse all flags?